src/activation.rs

```diff
@@ -142,11 +152,89 @@ pub(crate) fn get_activator<'p>(
     Ok(activator)
 }
 
+/// Get the environment variables from the shell environment.
+/// This method will get the environment variables from the shell environment and return them as a HashMap.
+/// If the environment variable does not exist, an empty string will be used.
+fn get_environment_variable_from_shell_environment(names: Vec<&String>) -> HashMap<String, String> {
+    names
+        .into_iter()
+        .map(|name| {
+            let value = std::env::var(name).unwrap_or_default();
```

Comment:
Do you really need to cache environment variables that are not on the system?

If yes, an `None` would be nicer than an empty string

Reply:
Good point, actually showed a bug as unset and empty values would become the same. 
Fixed in [`32033d4` (#2367)](https://github.com/prefix-dev/pixi/pull/2367/commits/32033d4aa0870f1f85586c7d80bb6e5a9d4f015f)